### PR TITLE
Introducing tests for rich-text parsing

### PR DESCRIPTION
**Pull request by goofballLogic**
**N.B. All the files in /test/lib are part of the jasmine framework and don't need to be code reviewed.**

As mentioned in the blog, this PR adds the Jasmine testing framework so that I can make sure parsing of rich text snippets is done correctly.

### CODE REVIEW

File: js/lib/node-artist/rich-text-processor.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import parser from "../../lib/rich-text/parser.js";
+
+function mapContent(data, element, expression) {
+
+    const parsed = parser(data[expression]);
+    element.appendChild(parsed);
+
+}
+
+export default function(node, data) {
```
Comment (YorgunKelebek) [inline js/lib/node-artist/rich-text-processor.js]: I thought we had agreed to name the default functions, for consistency?

File: js/lib/rich-text/parser.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default function(content) {
+
+    const buffer = document.createRange();
```
Comment (YorgunKelebek) [inline js/lib/rich-text/parser.js]: Need to read about these as I am not familiar with them...

Comment (goofballLogic) [reply js/lib/rich-text/parser.js]: https://developer.mozilla.org/en-US/docs/Web/API/Range

File: js/lib/node-artist/rich-text-processor.js (hunk 1 above)
Comment (goofballLogic) [reply js/lib/node-artist/rich-text-processor.js]: Thanks fixed

**Pull request merged**